C/src/TaskSetGenerator/BasicTaskSetGeneratorTemplate.lf

```diff
@@ -0,0 +1,135 @@
+/**
+ * Taskset Generator of Lingua Franca
+ * @author Hokeun Kim
+ * @author Yunsang Cho
+ */
+
+target C {
+    timeout: TOTAL_TIME,
+    workers: NUM_WORKERS,
+    scheduler: SCHEDULER_TYPE
+};
+
+preamble {=
+    typedef struct task_config_t {
+        int id;
+        long long int release_time;
+        long long int exe_time;
+        bool    periodic;
+        long long int period;
+    } task_config_t;
+    //#define 
+=}
+
+// Each Task has information about its release time and exeuction time
+reactor Task {
+
+    input in:task_config_t;
+    state exe_time:time(0 sec);
+    state id:int;
+    state periodic:bool(false);
+    state period:time(0 sec);
+    logical action release;
+
+    reaction(startup) {=
+        #ifdef TASK_SET_TRACING_ON
+            if (!register_user_trace_event("ID")) {
+                fprintf(stderr, "ERROR: Failed to register trace event.\n");
+                exit(1);
+            }
+        #endif // TASK_SET_TRACING_ON
+    =}
+
+    reaction(in) -> release {=
+        self->id = in->value.id;
+        self->exe_time = in->value.exe_time;
+        self->periodic = in->value.periodic;
+        self->period = in->value.period;
+        lf_schedule(release, in->value.release_time);
+    =}
+    
+    reaction(release) -> release {=
+        long long int physical_start_time = lf_time_physical();
+        //tracepoint_user_value("ID", self->id);
+        printf("Task %d released at logical time %lld nsec, physical time %lld nsec, execution time %lld nsec\n",
+            self->id,
+            lf_time_logical_elapsed(),
+            lf_time_physical_elapsed(),
+            self->exe_time
+        );
+        while (lf_time_physical() < physical_start_time + self->exe_time) {
+
+        };
+        printf("Task %d finished execution at physical time %lld nsec\n",
+            self->id,
+            lf_time_physical_elapsed());
+        if (self->periodic) {
+            lf_schedule(release, self->period);
+        }
```

Comment:
I think in the case of sporadic, it could use a random period here (with `self->period` being the maximum (inter-arrival time)).

Reply:
I like the way you suggest it!
Your idea is closer to the definition of "sporadic task" than it is to the way it is now.
I will change the code with your feedback!